### ISSUE TITLE
Set Browser Implicit Wait weird behavior

### ISSUE THREAD

**Issue by leandrolr90** [robotframework/SeleniumLibrary#1350]
* Using the keyword Set Browser Implicit Wait will set the implicit wait but it can not be read with Get Selenium Implicit Wait, which returns only the value set with Set Selenium Implicit Wait or when importing the library.
* When you open a new tab (with js for exemple) and go back to the main tab the implicit wait is gone but it will read the last value that have been set with Set Selenium Implicit Wait.

## Steps to reproduce the issue
```
*** Test Cases *** 
   # Open the browser and stuff...
    Log implicit wait    before
    Set Browser Implicit Wait    3s
    Log implicit wait    after set browser
    Set Selenium Implicit Wait    2s
    Log implicit wait    after set 2s
    Execute Javascript    window.open("about:blank")
    Select Window    NEW
    Select Window    MAIN
    # this will output 'after tab switch = 2 seconds' but the real value will be 0.0
    Log implicit wait    after tab switch

*** Keyword ***
Log implicit wait
    [Arguments]    ${prefix}
    ${implicit_wait} =    Get Selenium Implicit Wait
    Log    ${prefix} = ${implicit_wait}    console=True
```

## Environment

Browser: Google Chrome 73.0.3683.75
Browser driver: chromedriver=2.44.609538
Operating System: Windows 10 x64
Libraries
- Robot Framework: 3.1
- Selenium: 3.141.0
- SeleniumLibrary: 3.3.1
- Interpreter: Python 3.7.0b4 on win32

**Comment by aaltat** [comment]
Should the issue be reproduced if I add the `Open Browser` keyword in the start of the test?

So there are two problems. 
1) Setting the implicit wait and getting the implicit wait do not return same value
2) When opening new window with JavaScript, the implicit wait is lost from the browser.

**Comment by leandrolr90** [comment]
A new browser instance will have de default value for implicit wait (from the import or the last value from `Set Selenium Implicit Wait`).
When a new tab is open with a link `target="_blank"` the implicit is lost too.
If you use `Set Selenium Implicit Wait` and switch tabs the implicit is lost, even if you didn't mess with `Set Browser Implicit Wait`.
It seens that if you set implicit wait with keywords then switch tabs, the implicit will be lost but it reads the last value set with `Set Selenium Implicit Wait`.

Should i change the title?

**Comment by aaltat** [comment]
Could you tell what:
```
If you use Set Selenium Implicit Wait and switch tabs the implicit is lost
```
The implicit wait is lost from the browser or somewhere else ( if yes, from where)? How did you verify that it's lost?

The SeleniumLibrary set and get the correct implicit wait value? Only when switch between tabs, the browser does lost the implicit wait?

Title can remain for now.

**Comment by leandrolr90** [comment]
It's lost from the current browser (the one that switched tabs), other browsers instances will respect last value from `Set Selenium Implicit Wait`. You can tell that is lost because selenium keywords execute with no wait after that (can check elapsed time in report log too).

So, to resume:

Selenium sets implicit wait correctly but can't get and use the value after a tab switch. It affects only the browser that have switched tabs, a new browser instance or an already opened one that didn't switched tabs get and use the right value ( can tell this logging the value from `Get Selenium Implicit Wait` and checking report for the elapsed time of selenium keywords ).

**Comment by aaltat** [comment]
Ok, it's clear. Have you tried with Firefox? To rule out that this is not a Chrome specific problem?

**Comment by leandrolr90** [comment]
Will test, update in a minute.

It affects firefox too, the browser in which I switched tabs with `Select Window` lost the implicit wait. With an observation that in firefox the js `window.open("about:blank")` or `window.open()` opens a new separate window but stills get affected. 

```
Library    SeleniumLibrary    implicit_wait=2s

    Open browser    ${myurl}    firefox    alias=first
    # logs 'before = 2 seconds', ok    
    Log implicit wait    before
    # inputs text, clicks.. to check in report
    Login

    Open browser    ${myurl}    firefox
    Set Selenium Implicit Wait    5s
    Log implicit wait    after set 5s
    # implicit working
    Login
    Execute Javascript    window.open()
    Select Window    NEW
    Select Window    MAIN
    # logs 'after select window = 5 seconds', but the real is 0.0
    Log implicit wait    after select window
    Use selenium keywords

    Switch Browser    first
    # implicit is normal here
    Do selenium stuff
    # logs 'first = 5 seconds'
    Log implicit wait    first
```

**Comment by aaltat** [comment]
Thanks for the report, I need to investigate and see where the problem actually is.

**Comment by aaltat** [comment]
I did notice that implicit wait is not that well tested, so I did write new tests for it. But I could not reproduce your problem. Could you look at my test cases, in PR #1360 and can use the test to reproduce the problem?

**Comment by leandrolr90** [comment]
I used your test cases and could not reproduce the problem too, and now I see it don't exist. Then I tested it in the web app that I'm working, it has a page which opens in a new windows when you navigate to it, the "problem" was that in this page, elements were never ready at the first try, so selenium always implicitly waited, while in others pages it worked without the need of waiting.
Sorry to make you waste your time, my fault.
